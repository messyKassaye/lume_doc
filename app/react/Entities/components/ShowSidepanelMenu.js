/** @format */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ShowIf from 'app/App/ShowIf';
import { Icon } from 'UI';

export class ShowSidepanelMenu extends Component {
  render() {
    const { panelIsOpen, openPanel, active } = this.props;
    return (
      <div className={active ? 'active' : ''}>
        <ShowIf if={!panelIsOpen}>
          <div className="btn btn-primary" onClick={e => openPanel(e)}>
            <Icon icon="chart-bar" />
          </div>
        </ShowIf>
      </div>
    );
  }
}

ShowSidepanelMenu.defaultProps = {
  active: false,
};

ShowSidepanelMenu.propTypes = {
  active: PropTypes.bool,
  panelIsOpen: PropTypes.bool.isRequired,
  openPanel: PropTypes.func.isRequired,
};
