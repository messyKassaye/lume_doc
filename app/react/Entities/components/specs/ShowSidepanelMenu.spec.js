import React from 'react';

import { shallow } from 'enzyme';

import { ShowSidepanelMenu } from '../ShowSidepanelMenu';

describe('ShowSidepanelMenu', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      active: false,
      panelIsOpen: true,
      openPanel: jest.fn(),
    };
  });

  function render() {
    component = shallow(<ShowSidepanelMenu {...props} />);
  }

  beforeEach(() => {
    render();
  });

  it('should show button if panel is not open', () => {
    props.panelIsOpen = false;
    render();
    expect(component).toMatchSnapshot();
  });

  it('should call openPanel when button is clicked', () => {
    props.panelIsOpen = false;
    render();
    component.find('.btn').simulate('click');
    expect(props.openPanel).toHaveBeenCalled();
  });
});
