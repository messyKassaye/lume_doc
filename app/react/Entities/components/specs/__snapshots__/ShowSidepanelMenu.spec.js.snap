// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowSidepanelMenu should show button if panel is not open 1`] = `
<div
  className=""
>
  <ShowIf
    if={true}
  >
    <div
      className="btn btn-primary"
      onClick={[Function]}
    >
      <Connect(Icon)
        icon="chart-bar"
      />
    </div>
  </ShowIf>
</div>
`;
